### PR TITLE
feat: [AXM-252] add settings for edx-ace push notifications

### PR DESCRIPTION
**Pull request by NiedielnitsevIvan**
feat: [AXM-252] add settings for edx-ace push notifications

### CODE REVIEW

File: lms/djangoapps/mobile_api/notifications/views.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            "date_created": "2024-04-18T07:39:37.132787Z",
+            "cloud_message_type": "FCM",
+            "application_id": "my_app_id"
+        }
+        ```
+    """
+
+    def create(self, request, *args, **kwargs):
+        if not getattr(settings, 'PUSH_NOTIFICATIONS_SETTINGS', None):
+            return Response('Push notifications are not configured.', status.HTTP_501_NOT_IMPLEMENTED)
```
Comment (monteri) [inline lms/djangoapps/mobile_api/notifications/views.py:48]: 👍

File: openedx/core/djangoapps/ace_common/settings/common.py [hunk 2]
```diff
@@ -22,3 +25,29 @@ def plugin_settings(settings):  # lint-amnesty, pylint: disable=missing-function
     settings.ACE_ROUTING_KEY = ACE_ROUTING_KEY
 
     settings.FEATURES['test_django_plugin'] = True
+    settings.FCM_APP_NAME = 'fcm-edx-platform'
+
+    if getattr(settings, 'FIREBASE_SETTING_UP', None) is None:
```
Comment (monteri) [inline openedx/core/djangoapps/ace_common/settings/common.py:30]: This variable name means Firebase setup status? Maybe something like `FIREBASE_SETUP_STATUS` will work here?

File: openedx/core/djangoapps/ace_common/utils.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+"""
+Utility functions for edx ace.
```
Comment (monteri) [inline openedx/core/djangoapps/ace_common/utils.py:2]:
```suggestion
Utility functions for edx-ace.
```

File: openedx/core/djangoapps/ace_common/utils.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+"""
+
+
+def setup_firebase_app(firebase_credentials, app_name='fcm-app'):
+    """
+    Returns a Firebase app instance if the Firebase credentials are provided.
+    """
+    try:
+        import firebase_admin  # pylint: disable=import-outside-toplevel
+    except ImportError:
```
Comment (monteri) [inline openedx/core/djangoapps/ace_common/utils.py:12]: Maybe log here with text: "Could not import firebase_admin package"?

File: openedx/core/djangoapps/notifications/policies.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            return PolicyResult(deny={ChannelType.PUSH})
+
+        course_keys = [CourseKey.from_string(course_id) for course_id in course_ids]
+        for course_key in course_keys:
+            course_notif_preference = CourseNotificationPreference.get_user_course_preference(
+                message.recipient.lms_user_id,
+                course_key
+            )
+
+            if not course_notif_preference.get_notification_type_config(app_name, notification_type).get('push', False):
```
Comment (monteri) [inline openedx/core/djangoapps/notifications/policies.py:35]:
```suggestion
            if not course_notification_preference.get_notification_type_config(app_name, notification_type).get('push', False):
```

Comment (monteri) [reply openedx/core/djangoapps/notifications/policies.py:35]: Maybe create a variable before the if statement to reduce the condition size and provide more clarity here

File: openedx/core/djangoapps/ace_common/settings/common.py (hunk 2 above)
Comment (NiedielnitsevIvan) [reply openedx/core/djangoapps/ace_common/settings/common.py]: Yep, renamed.

File: openedx/core/djangoapps/ace_common/utils.py (hunk 3 above)
Comment (NiedielnitsevIvan) [reply openedx/core/djangoapps/ace_common/utils.py]: Fixed.

File: openedx/core/djangoapps/ace_common/utils.py (hunk 4 above)
Comment (NiedielnitsevIvan) [reply openedx/core/djangoapps/ace_common/utils.py:15]: Logs added.

File: openedx/core/djangoapps/notifications/policies.py (hunk 5 above)
Comment (NiedielnitsevIvan) [reply openedx/core/djangoapps/notifications/policies.py]: Done

**Pull request merged**